mmrazor/models/losses/dafl_loss.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from mmcv.runner import get_dist_info
+
+from mmrazor.registry import MODELS
+from ..ops import GatherTensors
+
+
+class DAFLLoss(nn.Module):
+    """Base class for DAFL losses.
+
+    paper link: https://arxiv.org/pdf/1904.01186.pdf
+
+    Args:
+        loss_weight (float): Weight of the loss.
+    """
+
+    def __init__(self, loss_weight=1.0) -> None:
+        super().__init__()
+        self.loss_weight = loss_weight
+
+    def forward(self, preds_T: torch.Tensor) -> torch.Tensor:
+        """Forward function for the DAFLLoss.
+
+        Args:
+            preds_T (torch.Tensor): The predictions of teacher.
+        """
+        return self.loss_weight * self.forward_train(preds_T)
+
+    def forward_trian(self, preds_T: torch.Tensor) -> torch.Tensor:
+        """Forward function during training.
+
+        Args:
+            preds_T (torch.Tensor): The predictions of teacher.
+        """
+        raise NotImplementedError
+
+
+@MODELS.register_module()
+class ActivationLoss(DAFLLoss):
+    """The loss function for measuring the activation of the target featuremap.
+    It is negative of the norm of the target featuremap.
+
+    Args:
+        norm_type (str, optional):The type of the norm. Defaults to 'abs'.
+    """
+
+    def __init__(self, norm_type='abs', **kwargs) -> None:
+
+        super().__init__(**kwargs)
+        assert norm_type in ['norm', 'abs'], \
+            '"norm_type" must be "norm" or "abs"'
+        self.norm_type = norm_type
+
+        if self.norm_type == 'norm':
+            self.norm_fn = lambda x: -x.norm()
+        elif self.norm_type == 'abs':
+            self.norm_fn = lambda x: -x.abs().mean()
+
+    def forward(self, feat_T: torch.Tensor) -> torch.Tensor:
```

Comment:
Is it exactly same func of its base class?

Reply:
Change its parent class to nn.Module.